src/main/java/com/google/alpollo/model/SongEntity.java

```diff
@@ -26,4 +35,21 @@ public int compare(SongEntity a, SongEntity b) {
       return Double.compare(b.salience, a.salience);
     }
   };
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == this) {
+      return true;
+    }
+
+    if (!(obj instanceof SongEntity)) { 
+      return false; 
+  } 
+
+    SongEntity entity = (SongEntity) obj;
+    return salience == entity.salience
+        && (name.equals(entity.name))
```

Comment:
`Objects.equals(name, entity.name)`, because it will fail with exception if `name == null`

Reply:
But name can never be null, it will always have a value. That's why I chose string.equals().

Reply:
Logically can't, but you don't stop me from doing this: `SongEntity newSongEntity = new SongEntity(null, 0.0, null, null);`

Reply:
Olya has a point here. You shouldn't trust your assumptions when designing APIs. In larger projects, we use annotations when we want to make sure a field is not null. For example, the constructor would be something like `public SongEntity(@NotNull String name, double salience, @Nullable String type, String wikiLink)`. That's not the case here, so we need to be prepared for someone passing `null`.

Also, in `wikiLink` documentation, you say that entities **might**  have one, so at least this field can be null, right?

Reply:
I made sure to set `wikiLink` to an empty string if it's ever null, but I got your point, `Object.equals()` just covers a wider range of cases and it's better for project safety.